lib/api/core/models/security/profile.js

```diff
@@ -21,23 +38,33 @@ function Profile () {
 Profile.prototype.isActionAllowed = function (requestObject, context, kuzzle) {
   var
     deferred = q.defer();
-
   if (this.roles === undefined || this.roles.length === 0) {
     return q(false);
   }
 
-  async.some(this.roles, (role, callback) => {
-    role.isActionAllowed(requestObject, context, kuzzle)
-      .then(isAllowed => {
-        callback(isAllowed);
+  this.getRoles(kuzzle)
+    .then(roles => {
+      async.some(roles, (role, callback) => {
+        role.isActionAllowed(requestObject, context, kuzzle)
+          .then(isAllowed => {
+            callback(isAllowed);
+          });
+      }, result => {
```

Comment:
Since `async` takes NodeJS-style callbacks, here you are processing an errored callback, and I believe this is not exactly what you wanted to code. :-)

Easy way to pass the result of a NodeJS callback to deferred promises:

```js
}, deferred.makeNodeResolver());
```